rust_src/src/window_configuration.rs

```diff
@@ -0,0 +1,203 @@
+            return false;
+        }
+
+        if !self.focus_frame.eq(other.focus_frame) {
+            return false;
+        }
+
+        true
+    }
+}
+
+impl SavedWindowRef {
+    pub fn equal(self, other: Self, ignore_positions: bool) -> bool {
+        // Windows' buffers must match.
+        if !self.buffer.eq(other.buffer) {
+            return false;
+        }
+
+        if !(self.pixel_left.eq(other.pixel_left)
+            && self.pixel_top.eq(other.pixel_top)
+            && self.pixel_height.eq(other.pixel_height)
+            && self.pixel_width.eq(other.pixel_width)
+            && self.left_col.eq(other.left_col)
+            && self.top_line.eq(other.top_line)
+            && self.total_cols.eq(other.total_cols)
+            && self.total_lines.eq(other.total_lines)
+            && self.display_table.eq(other.display_table))
+        {
+            return false;
+        }
+
+        // The next two check the window structure for equality.
+        if !(self.parent.eq(other.parent) && self.prev.eq(other.prev)) {
+            return false;
+        }
+
+        if !(ignore_positions
+            || (self.hscroll.eq(other.hscroll)
+                && self.min_hscroll.eq(other.min_hscroll)
+                && self.start_at_line_beg.eq(other.start_at_line_beg)
+                && equal(self.start, other.start))
+                && equal(self.pointm, other.pointm))
+        {
+            return false;
+        }
+
+        if !(self.left_margin_cols.eq(other.left_margin_cols)
+            && self.right_margin_cols.eq(other.right_margin_cols)
+            && self.left_fringe_width.eq(other.left_fringe_width)
+            && self.right_fringe_width.eq(other.right_fringe_width)
+            && self
+                .fringes_outside_margins
+                .eq(other.fringes_outside_margins)
+            && self.scroll_bar_width.eq(other.scroll_bar_width)
+            && self.scroll_bar_height.eq(other.scroll_bar_height)
+            && self
+                .vertical_scroll_bar_type
+                .eq(other.vertical_scroll_bar_type)
+            && self
+                .horizontal_scroll_bar_type
+                .eq(other.horizontal_scroll_bar_type))
+        {
+            return false;
+        }
+
+        true
+    }
+}
+
+impl From<LispObject> for SaveWindowDataRef {
+    fn from(o: LispObject) -> Self {
+        o.as_window_configuration_or_error()
+    }
+}
+
+impl LispObject {
+    pub fn as_window_configuration(self) -> Option<SaveWindowDataRef> {
+        self.as_vectorlike()
+            .and_then(|v| v.as_window_configuration())
+    }
+
+    pub fn as_window_configuration_or_error(self) -> SaveWindowDataRef {
+        self.as_window_configuration()
+            .unwrap_or_else(|| wrong_type!(Qwindow_configuration_p, self))
+    }
+}
+
+/// Return t if OBJECT is a window-configuration object.
+#[lisp_fn]
+pub fn window_configuration_p(object: LispObject) -> bool {
+    object.as_window_configuration().is_some()
```

Comment:
Why not use `is_window_configuration` ?

Reply:
One is not defined currently. Easy enough to add one.

Reply:
If you look at the function that was commented out it needs to make a Vectorlike object. I did not write a `is_window_configuration` because it felt less efficient than what I wrote above.